modules/cached-resources/post.js

```diff
@@ -0,0 +1,50 @@
+import { postObjectsToClass, postObjectsToClassWithRelation } from '../../services/parse/crud';
+import checkOnlineStatus from '../offline';
+import generateRandomID from '../utils';
+import {
+  storeData
+} from '../async-storage';
+
+function postIdentificationForm(postParams, setSurveyee, submitAction) {
+  checkOnlineStatus().then((connected) => {
+    if (connected) {
+      postObjectsToClass(postParams).then((surveyee) => {
+        const surveyeeSanitized = JSON.parse(JSON.stringify(surveyee));
+        setSurveyee(surveyeeSanitized);
+        submitAction();
+      });
+    } else {
+      const id = `PatientID-${generateRandomID()}`;
+      storeData(postParams, id);
+      submitAction();
+    }
+  });
+}
+
+function postSupplementaryForm(postParams, actions, toRoot) {
```

Comment:
so yea in the future this will be more akin to something like

```
function postSupplementaryForm(postParams) {
	checkOnlineStatus().then((connected) => {
		if (connected) {
		  postObjectsToClassWithRelation(postParams)
		}
		else {
			//All the fancy offline post logic here
		}
	})
}

```

but for now

```
function postSupplementaryForm(postParams) {
  postObjectsToClassWithRelation(postParams); 
}
```

it's stupid, but an important abstraction for now

you'll have to do this for all these functions

Reply:
should be good, converted all to promises